engines/scumm/he/net/net_main.h

```diff
@@ -0,0 +1,174 @@
+	int32 updateQuerySessions();
+	void stopQuerySessions();
+	int querySessions();
+	int queryProviders();
+	int setProvider(int providerIndex);
+	int closeProvider();
+	bool initAll();
+	bool initProvider();
+	bool initSession();
+	bool initUser();
+	void remoteStartScript(int typeOfSend, int sendTypeParam, int priority, int argsCount, int32 *args);
+	int remoteSendData(int typeOfSend, int sendTypeParam, int type, Common::String data, int priority, int defaultRes = 0, bool wait = false, int callid = 0);
+	void remoteSendArray(int typeOfSend, int sendTypeParam, int priority, int arrayIndex);
+	int remoteStartScriptFunction(int typeOfSend, int sendTypeParam, int priority, int defaultReturnValue, int argsCount, int32 *args);
+	void doNetworkOnceAFrame(int msecs);
+	void handleGameData(Common::JSONValue *json, int peerIndex);
+	void handleGameDataHost(Common::JSONValue *json, int peerIndex);
+
+private:
+	bool connectToSession(Common::String address, int port);
+	int doJoinSession(Session session);
+	bool serviceBroadcast();
+	void handleBroadcastData(Common::String data, Common::String host, int port);
+	void serviceSessionServer();
+	void handleSessionServerData(Common::String data);
+	bool remoteReceiveData(uint32 tickCount);
+
+public:
+	//getters
+	bool getHostName(char *hostname, int length);
+	bool getIPfromName(char *ip, int ipLength, char *nameBuffer);
+	void getSessionName(int sessionNumber, char *buffer, int length);
+	int getSessionPlayerCount(int sessionNumber);
+	void getProviderName(int providerIndex, char *buffer, int length);
+
+private:
+	//mostly getters
+	int getTotalPlayers();
+
+public:
+	//fields
+	int _latencyTime; // ms
+	bool _fakeLatency;
+
+	bool _isHost;  // true = hosting game, false = joined game.
+
+	int _myUserId;
+	int _fromUserId;
+
+	int _sessionId; // Session ID received from the session server.
+
+private:
+	ScummEngine_v90he *_vm;
+
+	Common::String _gameName;
+	Common::String _gameVersion;
+
+	Networking::ENet *_enet;
+
+	byte *_tmpbuffer;
+
+	int _numUsers;
+	int _numBots;
+	int _maxPlayers;
+	int _userIdCounter;
+	Common::HashMap<int, Common::String> _userIdToName;
+	Common::HashMap<int, int> _userIdToPeerIndex;
+	Common::HashMap<int, Common::String> _userIdToAddress;
+	Common::HashMap<Common::String, int> _addressToUserId;
+
+	Common::String _sessionName;
+	Networking::Host *_sessionHost;
+
+	bool _isShuttingDown;
+
+	Common::Queue<Common::JSONValue *> _hostDataQueue;
+	Common::Queue<int> _peerIndexQueue;
+
+	Common::Array<Session> _sessions;
+	int _hostPort;
+
+	// For broadcasting our game session over LAN.
+	Networking::Socket *_broadcastSocket;
+
+	// For creating/joining sessions over the Internet.
+	Networking::Host *_sessionServerHost;
+	Address _sessionServerAddress;
+	bool _forcedAddress;
+	bool _gotSessions;
+	int _sessionServerPeer;
+	bool _isRelayingGame; // If we're relaying in-game data over the session server or not.
+};
+
+} // End of namespace Scumm
+
+#endif
```

Comment:
Missing end of the line. I think there must be some setting in your editor to keep these.

Reply:
Just enabled the "Insert Final Newline" setting in Visual Studio Code, which I didn't know existed until just now.  This shouldn't be a problem anymore, thank you.